homeassistant/components/recollect_waste/calendar.py

```diff
@@ -0,0 +1,96 @@
+"""Support for ReCollect Waste calendars."""
+from __future__ import annotations
+
+import datetime
+
+from aiorecollect.client import PickupEvent
+
+from homeassistant.components.calendar import CalendarEntity, CalendarEvent
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import DOMAIN
+from .entity import ReCollectWasteEntity
+from .util import async_get_pickup_type_names
+
+
+@callback
+def async_get_calendar_event_from_pickup_event(
+    entry: ConfigEntry, pickup_event: PickupEvent
+) -> CalendarEvent:
+    """Get a HASS CalendarEvent from an aiorecollect PickupEvent."""
+    pickup_type_string = ", ".join(
+        async_get_pickup_type_names(entry, pickup_event.pickup_types)
+    )
+    return CalendarEvent(
+        summary="ReCollect Waste Pickup",
+        description=f"Pickup types: {pickup_type_string}",
+        location=pickup_event.area_name,
+        start=pickup_event.date,
+        end=pickup_event.date,
```

Comment:
Technically calendar end dates are a little strange and should be exclusive, so the end should be the next day (https://developers.home-assistant.io/docs/core/entity/calendar/#calendarevent)